modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/filter/ComplexFilterSplitter.java

```diff
@@ -121,7 +138,116 @@ public Object visit(Id filter, Object notUsed) {
         
         return null;
     }
-    
+
+    @Override
+    public Object visit(Function expression, Object notUsed) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        Object data = super.visit(expression, notUsed);
+        // encoding of functions with nested attributes as  arguments is not supported
+        if (nestedAttributes > 0 && preStack.size() == i + 1) {
+            Object o = preStack.pop();
```

Comment:
This code will never be hit. NestedAttributes is assigned as 0? Is it meant to be passed on to super.visit()?